### DISCUSSION TITLE
reexecuteQuery not using new variables

### DISCUSSION THREAD

**Discussion by bhalsodkhushal** [urql-graphql/urql#3469] [Q&A] [answered]
### Describe the bug

I am using urql version 4.0.5.

For fetching data i am calling below request using useQuery hook.

```
const [result, reexecuteQuery] = useQuery({
    query: GetUserQuery,
    variables: { userId: 1 },
});
```
For fetching new user data i am calling reexecuteQuery as per document, but it not working.

`reexecuteQuery({ variables: { userId: 3 } });`

### Reproduction

reexecuteQuery({ variables: { userId: 3 } });

### Urql version

 4.0.5

### Validations

- [X] I can confirm that this is a bug report, and not a feature request, RFC, question, or discussion, for which [GitHub Discussions](https://github.com/urql-graphql/urql/discussions) should be used
- [X] Read the [docs](https://formidable.com/open-source/urql/docs/).
- [X] Follow our [Code of Conduct](https://github.com/urql-graphql/urql/blob/main/CODE_OF_CONDUCT.md)

**Comment by kitten** [comment, marked as answer]
As per the TSDocs/TypeScript typings and our docs, `executeQuery` is _not_ a state setter and does not accept `variables`. It accepts an `OperationContext` object:

- https://urql.dev/goto/docs/basics/react-preact/#reexecuting-queries
- https://urql.dev/goto/docs/api/urql/#usequery

This is also not a pattern that makes much sense.
Either, you need to track your variables in state, and update that state, as it's an input to `useQuery`, or, if you're [doing programmatic queries, you can use the `Client` directly (e.g. via `useClient`)](https://urql.dev/goto/docs/basics/core/)

As per the issue template, I'm converting this to a discussion, since it's a usage issue/question

**Comment by bhalsodkhushal** [comment]
Okay, got it now. thanks
